UI/org.eclipse.birt.report.designer.ui/src/org/eclipse/birt/report/designer/internal/ui/views/SearchInputDialog.java

```diff
@@ -0,0 +1,599 @@
+
+		propCombo = new Combo(textContainer, SWT.BORDER | SWT.SINGLE /* | SWT.DROP_DOWN | SWT.READ_ONLY */);
+
+		gd = new GridData(GridData.GRAB_HORIZONTAL | GridData.HORIZONTAL_ALIGN_FILL);
+		gd.widthHint = 250;
+		propCombo.setLayoutData(gd);
+		propCombo.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				String text = propCombo.getText().trim();
+				if (text.length() == 0 || "(any)".equals(text)) {
+					setErrorMessage(Messages.getString("SearchInputDialog.Message.AllProps")); //$NON-NLS-1$
+				} else {
+					setErrorMessage(null);
+				}
+			}
+		});
+
+		errorMessageText = new Text(textContainer, SWT.READ_ONLY | SWT.WRAP);
+		gd = new GridData(GridData.GRAB_HORIZONTAL | GridData.HORIZONTAL_ALIGN_FILL);
+		errorMessageText.setLayoutData(gd);
+		errorMessageText.setBackground(errorMessageText.getDisplay().getSystemColor(SWT.COLOR_WIDGET_BACKGROUND));
+
+		Composite optionContainer = new Composite(composite, SWT.NONE);
+		gd = new GridData(SWT.FILL, SWT.NONE, false, false);
+		optionContainer.setLayoutData(gd);
+
+		ignoreCaseButton = new Button(optionContainer, SWT.CHECK);
+		Label ignoreCaseButtonLabel = new Label(optionContainer, SWT.SINGLE);
+		ignoreCaseButtonLabel.setText(Messages.getString("SearchInputDialog.Message.IgnoreCase")); //$NON-NLS-1$
+
+		wholeWordButton = new Button(optionContainer, SWT.CHECK);
+		Label wholeWordButtonLabel = new Label(optionContainer, SWT.SINGLE);
+		wholeWordButtonLabel.setText(Messages.getString("SearchInputDialog.Message.WholeWord")); //$NON-NLS-1$
+
+		regexButton = new Button(optionContainer, SWT.CHECK);
+		Label regexButtonLabel = new Label(optionContainer, SWT.SINGLE);
+		regexButtonLabel.setText(Messages.getString("SearchInputDialog.Message.RegularExpression")); //$NON-NLS-1$
+
+		recursiveButton = new Button(optionContainer, SWT.CHECK);
+		Label recursiveButtonLabel = new Label(optionContainer, SWT.SINGLE);
+		recursiveButtonLabel.setText(Messages.getString("SearchInputDialog.Message.Recursive")); //$NON-NLS-1$
+		recursiveButton.addSelectionListener(new SelectionListener() {
+
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				loadPropCombo();
+			}
+
+			@Override
+			public void widgetDefaultSelected(SelectionEvent e) {
+				loadPropCombo();
+			}
+		});
+
+		layout = new GridLayout();
+		layout.numColumns = 2;
+		layout.marginWidth = layout.marginHeight = 0;
+		optionContainer.setLayout(layout);
+
+		Composite tableContainer = new Composite(composite, SWT.NONE);
+		// gd = new GridData(GridData.GRAB_HORIZONTAL | GridData.HORIZONTAL_ALIGN_FILL);
+		gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gd.widthHint = 250;
+		gd.horizontalSpan = 2;
+		tableContainer.setLayoutData(gd);
+
+		layout = new GridLayout();
+		layout.numColumns = 1;
+		layout.marginWidth = layout.marginHeight = 0;
+		tableContainer.setLayout(layout);
+
+		final Table table = new Table(tableContainer, SWT.VIRTUAL | SWT.BORDER);
```

Comment:
SWT.FULL_SELECTIONwill make the selection be the full line which I think will look nicer.